sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -210,6 +210,29 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
             if (rx_info.flags & NETDEV_RX_IEEE802154_INFO_FLAG_TIMESTAMP) {
                 gnrc_netif_hdr_set_timestamp(hdr, rx_info.timestamp);
             }
+#endif
+#if !IS_ACTIVE(CONFIG_IEEE802154_AUTO_ACK_DISABLE)
+            if ((mhr[0] & IEEE802154_FCF_TYPE_MASK) == IEEE802154_FCF_TYPE_DATA &&
+                (mhr[0] & IEEE802154_FCF_ACK_REQ)) {
+                netopt_enable_t auto_ack = NETOPT_DISABLE;
+                netopt_enable_t promiscuous = NETOPT_DISABLE;
+                if (dev->driver->get(dev, NETOPT_PROMISCUOUSMODE, &promiscuous, sizeof(promiscuous)) > 0 &&
+                    dev->driver->get(dev, NETOPT_AUTOACK, &auto_ack, sizeof(auto_ack)) > 0 &&
+                    auto_ack != NETOPT_ENABLE && promiscuous != NETOPT_ENABLE) {
```

Comment:
Why is sending the ACK done in `gnrc_netif_ieee802154` instead of the submac?

Reply:
seemed to be the right place to me because then
- all needed information is there
- ACK is only sent after the security check passed
- also works for drivers not supporting submac.

Reply:
> frame does not pass the security layer
- might have a wrong MIC
- no key found to decrypt the frame
- frame is a replay

As I said, I dont find a clear answer whether ACK transmission happens after or before the security check.
I doubt that an adversary gains anything when he can trigger an ACK for invalid frames. Sending the ACK costs a bit of energy but not more than for the adversary.
So your point also makes sense to me that the ACK stops retransmissions for frames that were actually supposed to be valid from good nodes. But an adversary could as well ignore the ACK and keep sending.

Clearly, in the code it can be easily moved before or after.

Reply:
The thing is I'd really keep the mac layer functionality in the submac and not concern GNRC with it. Moving the l2 ACK sending into GNRC seems like an unnecessary layering violation.

When you keep it in the submac, you also don't have to expose `IEEE802154_CAP_AUTO_ACK` via `NETOPT_AUTOACK` which is rather a hack.

Reply:
I think here is the place to add it: https://github.com/fabian18/RIOT/blob/ed44a434641ff0516a01fea885c5e32625beb558/drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c#L267